exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/WebUtils.java

```diff
@@ -48,4 +68,87 @@ public static String generateCsrfToken() {
     new SecureRandom().nextBytes(buffer);
     return Base64.getUrlEncoder().withoutPadding().encodeToString(buffer);
   }
+
+  /**
+   * Build an URL of a remote Drillbit endpoint.
+   *
+   * @param work {@link WorkManager} instance needed to retrieve the Drillbit address.
+   * @param request {@link HttpServletRequest} instance needed to to set the URL schema.
+   * @param hostname hostname of the Drillbit.
+   * @param path relative path to the endpoint.
+   * @return remote Drillbit endpoint URL.
+   * @throws RuntimeException if there is no Drillbit at the given hostname.
+   */
+  static URL getDrillbitURL(WorkManager work, HttpServletRequest request, String hostname, String path) {
+    DrillbitEndpoint drillbit = work.getContext().getAvailableBits().stream()
+        .filter(db -> db.getAddress().equals(hostname))
+        .findAny()
+        .orElse(null);
+    if (drillbit == null) {
+      throw new RuntimeException(String.format("No such drillbit: %s", hostname));
+    }
+    URL url;
+    try {
+      url = new URL(request.getScheme(), hostname, drillbit.getHttpPort(), path);
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
+    return url;
+  }
+
+  /**
+   * Send an HTTP request and returns response body as String.
+   *
+   * @param httpRequest {@link org.apache.http.client.methods.HttpGet} or {@link org.apache.http.client.methods.HttpPost} instance representing an HTTP request.
+   * @param drillConfig {@link DrillConfig} instance needed to setup HTTP Client.
+   * @return String response body.
+   * @throws Exception if unable to create HTTP client or in case of HTTP timeout.
+   */
+  static String doHTTPRequest(HttpRequestBase httpRequest, DrillConfig drillConfig) throws Exception {
+    CloseableHttpAsyncClient httpClient = getHttpClient(drillConfig);
```

Comment:
Reasonable timeout? Give Drillbit 2 seconds to respond, else display "Unavailable"?